packages/flutter/lib/src/services/platform_views.dart

```diff
@@ -0,0 +1,160 @@
+  /// which creates the platform view and manages its lifecycle.
+  int getNextPlatformViewId() => _nextPlatformViewId++;
+}
+
+/// Callback signature for when a platform view was created.
+///
+/// `id` is the platform view's unique identifier.
+typedef void OnPlatformViewCreated(int id);
+
+/// Provides access to the platform views service.
+///
+/// This service allows creating and controlling Android views.
+///
+/// See also: [PlatformView].
+class PlatformViewsService {
+  PlatformViewsService._();
+
+  /// Creates a controller for a new Android view.
+  ///
+  /// `id` is an unused unique identifier generated with [platformViewsRegistry].
+  ///
+  /// `viewType` is the identifier of the Android view type to be created, a
+  /// factory for this view type must have been registered on the platform side.
+  /// Platform view factories are typically registered by plugin code.
+  /// Plugins can register a platform view factory with
+  /// [PlatformViewRegistry#registerViewFactory](/javadoc/io/flutter/plugin/platform/PlatformViewRegistry.html#registerViewFactory-java.lang.String-io.flutter.plugin.platform.PlatformViewFactory-).
+  ///
+  /// The Android view will only be created after [AndroidViewController.setSize] is called for the
+  /// first time.
+  static AndroidViewController initAndroidView({
+    @required int id,
+    @required String viewType,
+    OnPlatformViewCreated onPlatformViewCreated,
+  }) {
+    return new AndroidViewController._(
+        id: id,
+        viewType: viewType,
+        onPlatformViewCreated: onPlatformViewCreated
+    );
+  }
+}
+
+enum _AndroidViewState {
+  WAITING_FOR_SIZE,
+  CREATING,
+  CREATED,
+  CREATE_FAILED,
+  DISPOSED,
+}
+
+/// Controls an Android view.
+///
+/// Typically created with [PlatformViewsService.initAndroidView].
+class AndroidViewController {
+  AndroidViewController._({
+    @required this.id,
+    @required String viewType,
+    OnPlatformViewCreated onPlatformViewCreated,
+  }) : assert(id != null),
+       assert(viewType != null),
+       _viewType = viewType,
+        _onPlatformViewCreated = onPlatformViewCreated,
+       _state = _AndroidViewState.WAITING_FOR_SIZE;
+
+  /// The unique identifier of the Android view controlled by this controller.
+  final int id;
+
+  final String _viewType;
+
+  final OnPlatformViewCreated _onPlatformViewCreated;
+
+  /// The texture entry id into which the Android view is rendered.
+  int _textureId;
+
+  /// Returns the texture entry id that the Android view is rendering into.
+  ///
+  /// Returns null if the Android view has not been successfully created, or if it has been
+  /// disposed.
+  int get textureId => _textureId;
+
+  _AndroidViewState _state;
+
+  /// Disposes the Android view.
+  ///
+  /// The [AndroidViewController] object is unusable after calling this.
+  /// The identifier of the platform view cannot be reused after the view is
+  /// disposed.
+  Future<Null> dispose() async {
```

Comment:
Nit: I think these should all be `Future<void>`

Reply:
Thanks, I didn't know we can do `Future<void>`, what's the difference between `Future<Null>` and `Future<void>` ?

`Future<Null>` seems to be more common in the framework codebase (~100 `Future<Null>`s vs ~10 `Future<void>`)